### PR TITLE
[SPARK-30234][SQL][FOLLOWUP] Add `.enabled` in the suffix of the ADD FILE legacy option

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This pr intends to rename `spark.sql.legacy.addDirectory.recursive` into `spark.sql.legacy.addDirectory.recursive.enabled`.

### Why are the changes needed?

For consistent option names.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

N/A

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -2116,11 +2116,13 @@ object SQLConf {
...
       .createWithDefault(false)
 
-  val LEGACY_ADD_DIRECTORY_USING_RECURSIVE = buildConf("spark.sql.legacy.addDirectory.recursive")
-    .doc("When true, users can add directory by passing path of a directory to ADD FILE " +
-      "command of SQL. If false, then only a single file can be added.")
-    .booleanConf
-    .createWithDefault(true)
+  val LEGACY_ADD_DIRECTORY_USING_RECURSIVE =
+    buildConf("spark.sql.legacy.addDirectory.recursive.enabled")
+      .internal()
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: We need `internal()` for legacy options? Currently, some of the legacy options have `internal()` and the others have not. Any consistent policy for that?

**Comment by maropu** [comment]
cc: @gatorsmile @HyukjinKwon

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 1 above)
Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I don't think we have such thing. Seems like it has been just case-by-case.

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Review by dongjoon-hyun — COMMENTED** [review]
Please change the following together in `DDLSuite.scala` for consistency.
```scala
test("Add a directory when spark.sql.legacy.addDirectory.recursive set to true")
test("Add a directory when spark.sql.legacy.addDirectory.recursive not set to true") {
```

**Comment by maropu** [comment]
sure

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**